File: backends/go-gin/internal/domain/invitation/repository.go
package invitation

// Repository は招待の永続化インターフェースです。
type Repository interface {
	// GetCurrentByRole はロールで絞り込んだ最新の招待情報の値オブジェクトを返します。
	GetCurrentByRole(role int) (*Vo, error)
	// Issue は新しい招待トークンを生成して保存し、値オブジェクトを返します。
	Issue(role int) (*Vo, error)
	// FindByToken はトークンで招待情報の値オブジェクトを返します。
	FindByToken(token string) (*Vo, error)
}

File: backends/kotlin-ktor/src/main/kotlin/com/authorization/domain/invitation/Repository.kt
/*
 * 招待ドメインリポジトリインターフェースモジュール。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
package com.authorization.domain.invitation

/**
 * 招待リポジトリのインターフェースです。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
interface Repository {

    /**
     * 指定ロールの現在有効な招待情報を取得します。
     *
     * @param role ロール（1=管理者、2=メンバー）
     * @return 招待 VO、または null
     */
    suspend fun getCurrentByRole(role: Int): Vo?

    /**
     * 指定ロールで招待トークンを新規発行します。
     *
     * @param role ロール（1=管理者、2=メンバー）
     * @return 発行された招待 VO
     */
    suspend fun issue(role: Int): Vo

    /**
     * 招待トークンに一致する招待情報を取得します。
     *
     * @param token 招待トークン
     * @return 招待 VO、または null
     */
    suspend fun findByToken(token: String): Vo?
}
